diff --git a/node_modules/@modelcontextprotocol/sdk/dist/cjs/client/sse.js b/node_modules/@modelcontextprotocol/sdk/dist/cjs/client/sse.js
index 0a77250..9625596 100644
--- a/node_modules/@modelcontextprotocol/sdk/dist/cjs/client/sse.js
+++ b/node_modules/@modelcontextprotocol/sdk/dist/cjs/client/sse.js
@@ -59,6 +59,7 @@ class SSEClientTransport {
                     ...init,
                     headers: {
                         ...headers,
+                        ...(this._requestInit ? this._requestInit.headers : void 0),
                         Accept: "text/event-stream"
                     }
                 })),
diff --git a/node_modules/@modelcontextprotocol/sdk/dist/esm/client/sse.js b/node_modules/@modelcontextprotocol/sdk/dist/esm/client/sse.js
index 0ef0554..3285fd3 100644
--- a/node_modules/@modelcontextprotocol/sdk/dist/esm/client/sse.js
+++ b/node_modules/@modelcontextprotocol/sdk/dist/esm/client/sse.js
@@ -55,6 +55,7 @@ export class SSEClientTransport {
                     ...init,
                     headers: {
                         ...headers,
+                        ...(this._requestInit ? this._requestInit.headers : void 0),
                         Accept: "text/event-stream"
                     }
                 })),
